### ISSUE TITLE
Impossible to have multiple bitfields columns

### ISSUE THREAD

**Issue by MathieuDerelle** [grosser/bitfields#21]
It's possible to set up multiple columns and retrieve bits, but it's impossible to set the values on specific columns with same bitfields 

```
  bitfield :exposed_fields, 2**0 => :title,
                            2**1 => :last_name,
                            2**2 => :first_name,
                            2**3 => :address_1,
                            2**4 => :address_2,
                            2**5 => :postal_code,
                            2**6 => :city

  bitfield :required_fields, 2**0 => :title,
                             2**1 => :last_name,
                             2**2 => :first_name,
                             2**3 => :address_1,
                             2**4 => :address_2,
                             2**5 => :postal_code,
                             2**6 => :city
```

**Comment by grosser** [comment]
so rename them ?

```Ruby
bitfield :exposed_fields, 2**0 => :exposed_title
bitfield :required_fields, 2**0 => :required_title
```

**Comment by dan-ding** [comment]
Rails enums have the same issue. It's due to generated methods and scopes.

What if there were an option not to create/use them?

**Comment by grosser** [comment]
might as well create unique names ?
... not creating them does not make much sense ... then just don't use bitfields and set the bits manually ?

**Comment by dan-ding** [comment]
Good point on not making them, Creating them different on purpose seems the solution (and is the same with Rails enums).

**Comment by take-cheeze** [comment]
Having a prefix or suffix like rails enum would be great.

**Comment by grosser** [comment]
all the code relies on unique names ... so idk how to make this work / how devs would interact with the resulting records ... if you want a prefix then just write it out ... straight forward and findable/grepable later ... or dynamically generate the config hash ...

**Comment by take-cheeze** [comment]
OK. Seems like this gem is for adding attributes for the row, not the column.
So should add another table for a correct use.

Anyway this gem is still very useful. Thanks for your codes!

**Comment by grosser** [comment]
can look into how ActiveRecord enum does it and if that fits better :)
(for the whole changes/accessors code it makes little sense to have duplicate keys)

**Comment by pboling** [comment]
Note: multiple flag columns per model, and multi-bit-per-flag (ternary/enum) columns, are supported in the alternative project `flag_shih_tzu` (https://github.com/galtzo-floss/flag_shih_tzu).
